packages/scandipwa/src/route/UrlRewrites/UrlRewrites.container.tsx

```diff
@@ -59,9 +60,10 @@ export class UrlRewritesContainer extends PureComponent<UrlRewritesContainerProp
     initialUrl = '';
 
     componentDidMount(): void {
-        if (this.getIsLoading()) {
-            this.requestUrlRewrite();
-        }
+        waitForPriorityLoad().then(
+            /** @namespace Route/UrlRewrites/Container/UrlRewritesContainer/componentDidMount/waitForPriorityLoad/then */
+            () => this.requestUrlRewrite(),
+        );
```

Comment:
I don't get why this is necessary, how does it fix the issue and what is the actual root cause of it?

Reply:
well this `() => this.requestUrlRewrite(),` wasnt runing at all on first loads (direct links) but working noramly on navigation what cause some variable not correctly set which leads to continuous placholders (note: those palceholders are running conditionally not fallback

Reply:
`urlRewrite` data should be preloaded in the document and it shouldn't be necessary to make a GraphQl request for it

Reply:
yeah but not all the needed data are preloaded looks like this ticket will need more updates (backend changes I mean)